### ISSUE TITLE
Controller's menuIsShowing is not getting updated.

### ISSUE THREAD

**Issue by cs-vladpaianu** [malikwang/custom_pop_up_menu#10]
First of all, hello and a big congrats for this awesome package! 

It fits the exact needs I have for my project, but I have encountered a problem with it. Unless you explicitly call "controller.showMenu()" or "controller.hideMenu" or "controller.toggleMenu()", the **controller.menuIsShowing** property never gets updated. 

It should be automatically set to true when the menu appears in my opinion, because outside of the component we cannot know if it is true or not, as we cannot wrap the child component of CustomMenuPopup with a GestureDetector, as the menu would not show anymore.

Thank you and I am looking forward to a reply on this matter.

**Comment by malikwang** [comment]
> First of all, hello and a big congrats for this awesome package!
> 
> It fits the exact needs I have for my project, but I have encountered a problem with it. Unless you explicitly call "controller.showMenu()" or "controller.hideMenu" or "controller.toggleMenu()", the **controller.menuIsShowing** property never gets updated.
> 
> It should be automatically set to true when the menu appears in my opinion, because outside of the component we cannot know if it is true or not, as we cannot wrap the child component of CustomMenuPopup with a GestureDetector, as the menu would not show anymore.
> 
> Thank you and I am looking forward to a reply on this matter.

You can init a controller outside, and pass it to CustomMenuPopup.
plz provide the code snippet for me to understand this problem better.

**Comment by cs-vladpaianu** [comment]
```
import 'package:flutter/material.dart';
import 'package:custom_pop_up_menu/custom_pop_up_menu.dart';

class Screen extends StatefulWidget {
  final Future<List<String>> stringList;

  Screen({this.stringList});
  @override
  _ScreenState createState() => _ScreenState();
}

class _ScreenState extends State<Screen> {
  CustomPopupMenuController _popupMenuController = CustomPopupMenuController();

  Widget _getPageContent() {
    return Column(
      mainAxisAlignment: MainAxisAlignment.start,
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        FutureBuilder<List<String>>(
          future: widget.stringList,
          builder: (context, snapshot) {
            if (snapshot.hasData && !snapshot.hasError) {
              List<String> stringList = snapshot.data;
              **// I tried instantiating a CustomMenuController here for every list element, but it did not work**
              return Container(
                height: stringList.length.toDouble() * 64,
                child: ListView.builder(
                    itemCount: stringList.length,
                    itemBuilder: (BuildContext context, int index) {
                      return Column(
                          mainAxisAlignment: MainAxisAlignment.center,
                          children: [
                            Container(
                              height: 64,
                              color: Colors.black,
                              child: Row(
                                mainAxisAlignment: MainAxisAlignment.start,
                                crossAxisAlignment: CrossAxisAlignment.center,
                                children: [
                                  SizedBox(width: 18),
                                  CustomPopupMenu(
                                    child: Container(
                                      height: 32,
                                      decoration: BoxDecoration(
                                          border: Border.all(
                                            **// HERE IS MY PROBLEM!!**
                                            color: _popupMenuController
                                                    .menuIsShowing
                                                ? Colors.red
                                                : Colors.transparent,
                                          ),
                                          borderRadius: BorderRadius.all(
                                              Radius.circular(8))),
                                      child: Padding(
                                        padding: const EdgeInsets.symmetric(
                                            horizontal: 4.0),
                                        child: Image.asset(
                                            "assets/markers_dots.png",
                                            width: 20,
                                            height: 4,
                                            color: Colors.red),
                                      ),
                                    ),
                                    menuBuilder: () => ClipRRect(
                                      borderRadius: BorderRadius.circular(5),
                                      child: Container(
                                        color: Colors.black,
                                        width: 200,
                                        child: Column(
                                            crossAxisAlignment:
                                                CrossAxisAlignment.stretch,
                                            children: [
                                              GestureDetector(
                                                behavior:
                                                    HitTestBehavior.translucent,
                                                onTap: () {
                                                  print("Do something");
                                                },
                                                child: Container(
                                                  height: 50,
                                                  child: Padding(
                                                    padding: const EdgeInsets
                                                            .symmetric(
                                                        horizontal: 16.0),
                                                    child: Row(
                                                      mainAxisAlignment:
                                                          MainAxisAlignment
                                                              .start,
                                                      crossAxisAlignment:
                                                          CrossAxisAlignment
                                                              .center,
                                                      children: [
                                                        SizedBox(width: 8),
                                                        Text("Code snippet"),
                                                      ],
                                                    ),
                                                  ),
                                                ),
                                              ),
                                            ]),
                                      ),
                                    ),
                                    pressType: PressType.singleClick,
                                    verticalMargin: 5,
                                    horizontalMargin: 20,
                                    controller: _popupMenuController,
                                    arrowColor: Colors.black,
                                    barrierColor: Colors.transparent,
                                  ),
                                ],
                              ),
                            )
                          ]);
                    }),
              );
            } else {
              return SizedBox(height: 0);
            }
          },
        ),
        SizedBox(height: 16)
      ],
    );
  }

  @override
  Widget build(BuildContext context) {
    return _getPageContent();
  }
}
```

In the code above, whenever I would call _popupMenuController.menuIsShowing, it would be false, despite the menu being shown as I tap on that element.

**Comment by malikwang** [comment]
@cs-vladpaianu 
Thanks for your demo. This has fixed on version 1.2.2.
For your case, your should add these code to work:
```dart
List<CustomPopupMenuController> _controllers = [];
List<String> stringList = snapshot.data;
if (_controllers.isEmpty) {
  stringList.forEach((_) {
    _controllers.add(CustomPopupMenuController());
  });
}
return Container(
  height: stringList.length.toDouble() * 64,
  child: ListView.builder(
    itemCount: stringList.length,
    itemBuilder: (BuildContext context, int index) {
      return Row(
        mainAxisAlignment: MainAxisAlignment.start,
        crossAxisAlignment: CrossAxisAlignment.center,
        children: [
          SizedBox(width: 18),
          CustomPopupMenu(
            child: Container(
              height: 32,
              decoration: BoxDecoration(
                border: Border.all(
                  color: _controllers[index].menuIsShowing ? Colors.red : Colors.transparent,
                ),
              ),
            ),
            // ****
            menuOnChange: (flag) {
              print(flag);
              setState(() {});
            },
            menuBuilder: () {
              // ...
            },
            controller: _controllers[index],

          ),
        ],
      );
    },
  ),
);
```
However, i think your case is similar with DropDownMenu, maybe you can try [mk_drop_down_menu](https://pub.dev/packages/mk_drop_down_menu).

**Comment by cs-vladpaianu** [comment]
Thank you very much for the change, it is working as expected now. I prefer this package to the other one, because this fits my design better. 

After updating to 1.2.2 though, the issue I have now is that when I call _controllers[index].hideMenu(), it is now not closing anymore. When a certain element in the menu in pressed, I want to show an AlertDialog and therefore hide the menu, but it currently is not dismissing. 

If I were to use it outside of a ListView, with a single controller, the hideMenu() method works. Do you have any idea what the issue might be? 

**EDIT**: Creating the controllersList empty list outside of the FutureBuilder did the trick apparently and everything is now working properly, thanks again!
